test/kubemark/start-kubemark.sh

```diff
@@ -446,55 +450,92 @@ function setup_proxy {
+  echo "DBG: image + tar files uploaded"
+  UPLOAD_TAR_DONE="done by kubemark"
+fi
+
 if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
   echo "DBG: Generating shared CA certificates"
   generate-shared-ca-cert
 
+  ### calculate the expected TENANT_SERVER_KUBECONFIGS and MASTER_METADATA,
+  #   which will be used by components that need to talk to all TPs.
+  export TENANT_SERVER_KUBECONFIGS
+  export MASTER_METADATA
+  for (( tp_num=1; tp_num<=${SCALEOUT_TP_COUNT}; tp_num++ ))
+  do
+      # TODO: fix the hardcoded path
+      # the path is what the controller used in master init script on the master machines
+      if [[ ${tp_num} == 1 ]]; then
+          export TENANT_SERVER_KUBECONFIGS="/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
+      else
+          export TENANT_SERVER_KUBECONFIGS="${TENANT_SERVER_KUBECONFIGS},/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
+      fi
+
+      if [[ ${tp_num} == 1 ]]; then
+          MASTER_METADATA="tp-${tp_num}=${TP_KUBECONFIG}-${tp_num}"
+      else
+          MASTER_METADATA=${MASTER_METADATA},"tp-${tp_num}=${TP_KUBECONFIG}-${tp_num}"
+      fi
+  done
+
   echo "DBG: Starting ${SCALEOUT_TP_COUNT} tenant partitions ..."
   export USE_INSECURE_SCALEOUT_CLUSTER_MODE="${USE_INSECURE_SCALEOUT_CLUSTER_MODE:-false}"
   export KUBE_ENABLE_APISERVER_INSECURE_PORT="${KUBE_ENABLE_APISERVER_INSECURE_PORT:-false}"
   export KUBERNETES_TENANT_PARTITION=true
   export KUBERNETES_RESOURCE_PARTITION=false
   export PROXY_KUBECONFIG
 
+  # the (shared) bearer token is needed for all TP/RP provisioning
+  export SHARED_APISERVER_TOKEN="$(secure_random 32)"
+  export KUBE_BEARER_TOKEN=${SHARED_APISERVER_TOKEN}
+  echo "DBG: shared bearer token: ${KUBE_BEARER_TOKEN}"
+
   for (( tp_num=1; tp_num<=${SCALEOUT_TP_COUNT}; tp_num++ ))
   do
-    export TENANT_PARTITION_SEQUENCE=${tp_num}
-    export KUBEMARK_CLUSTER_KUBECONFIG="${TP_KUBECONFIG}-${tp_num}"
-    create-kubemark-master
-    
-    export TP_${tp_num}_RESERVED_IP=$(cat ${KUBE_TEMP}/master_reserved_ip.txt)
-
-    # TODO: fix the hardcoded path
-    # the path is what the controller used in master init script on the master machines
-    if [[ ${tp_num} == 1 ]]; then
-      export TENANT_SERVER_KUBECONFIGS="/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
-    else
-      export TENANT_SERVER_KUBECONFIGS="${TENANT_SERVER_KUBECONFIGS},/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
-    fi
-
-    if [[ ${tp_num} == 1 ]]; then
-      MASTER_METADATA="tp-${tp_num}=${TP_KUBECONFIG}-${tp_num}"
-    else
-      MASTER_METADATA=${MASTER_METADATA},"tp-${tp_num}=${TP_KUBECONFIG}-${tp_num}"
-    fi
-
-    # export tp-1 token and share with other clusters
-    if [[ ${tp_num} == 1 ]]; then
-      tp1_token=$(grep token "${KUBEMARK_CLUSTER_KUBECONFIG}" | awk -F ": " '{print $2}')
-      export SHARED_APISERVER_TOKEN=${tp1_token}
-      echo "shares token: ${SHARED_APISERVER_TOKEN}"
-    fi
+    create_TP ${tp_num} &
```

Comment:
logs for multiple TP creation will be mixed, should have a TODO to track and fix

Reply:
will add a TODO line as reminder